### PR TITLE
Hypo can inject solutions back into beakers

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## About the PR

This allows Hypopens/Sprays to inject their solutions in beakers/anything that can carry the liquid

## Why / Balance

This was mostly for HypoSpray, right now if you put a bit too many chems in the Hypospray you just have to find something to inject, this can just let you empty your spray back in the jug.

~~This though buffs hypopens so that their solution can also be injected. I was unsure if this would be bad so I decided to wait for reviewers to tell me to potentially make a component to allow the old behavior.~~

nevermind

## Technical details

Nearly forgot to mention, since of ALL the comments below I added a new component to hypospray called `mobOnly` by default it's `true` which means everything will have the old behavior still. But in this pr I set the hypospray item to `false` which will let you inject into beakers.

If you think medipens should be able to inject themselves into beakers then just change the default value to false or add the component 

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Vasilis
- tweak: Hyposprays can inject their solutions back into beakers/jugs/whatever can hold the solution.

### CODE REVIEW

**Comment by GoodWheatley** [comment]
Doesn’t this allow emergency pens/space pens to inject into beakers?

**Comment by VasilisThePikachu** [comment]
> Doesn’t this allow emergency pens/space pens to inject into beakers?

Thats exactly what I mentioned, I kinda like this and dislike this

Think is... I don't see a reason to do that. Maybe med for the epinephrine

**Comment by Ilya246** [comment]
> > Doesn’t this allow emergency pens/space pens to inject into beakers?
> 
> Thats exactly what I mentioned, I kinda like this and dislike this
> 
> Think is... I don't see a reason to do that. Maybe med for the epinephrine

yeah, is there a real reason this shouldn't be something you can do?

**Comment by GoodWheatley** [comment]
> Think is... I don't see a reason to do that. Maybe med for the epinephrine

You can obtain barozine and make barozine pills through this.

**Comment by VasilisThePikachu** [comment]
> > > Doesn’t this allow emergency pens/space pens to inject into beakers?
> > 
> > 
> > Thats exactly what I mentioned, I kinda like this and dislike this
> > Think is... I don't see a reason to do that. Maybe med for the epinephrine
> 
> yeah, is there a real reason this shouldn't be something you can do?

Can you say why? But I will be making a component then like I mentioned

**Comment by Interrobang01** [comment]
i don't see a problem with barozine being obtainable

**Comment by VasilisThePikachu** [comment]
> > Think is... I don't see a reason to do that. Maybe med for the epinephrine
> 
> You can obtain barozine and make barozine pills through this.

Pills take time to metabolize, I don't see a reason to make pills for it. Just inject the pen

**Comment by GoodWheatley** [comment]
> > > Think is... I don't see a reason to do that. Maybe med for the epinephrine
> > 
> > 
> > You can obtain barozine and make barozine pills through this.
> 
> so? isn't it more convenient to just use the pen without beakering/pilling it

The material reclaimer had a similar issue when it was first released. Barozine is only supposed to be obtainable through space pens, and you shouldn’t be able to transfer it out.

**Comment by Ilya246** [comment]
> > > > Think is... I don't see a reason to do that. Maybe med for the epinephrine
> > > 
> > > 
> > > You can obtain barozine and make barozine pills through this.
> > 
> > 
> > so? isn't it more convenient to just use the pen without beakering/pilling it
> 
> The material reclaimer had a similar issue when it was first released. Barozine is only supposed to be obtainable through space pens, and you shouldn’t be able to transfer it out.

is there a reason as to why?

**Comment by Nairodian** [comment]
Pretty sure those chems can be abused to shit. The playerbase has found every other way to abuse a chem, I can see them doing it to that too. But I'm no maintainer.

**Comment by deltanedas** [comment]
recycler can already be used to extract chems from pens

**Comment by Equivocateur** [comment]
> Pretty sure those chems can be abused to shit. The playerbase has found every other way to abuse a chem, I can see them doing it to that too. But I'm no maintainer.

Leporazine can be done with plasma and has no overdose. It stabilizes temperature and heals cold rapidly.
Barozine cannot be made with chemdisp. It does 3 poison damage when over 30u plus vomiting.

Space pen has 20 barozine and 10 leporazine.

The only advantage over "just stab yourself with space pen lol" is more control over your dosage of barozine and thus being able to be in space for longer more consistently. Having barozine in a beaker only makes it easier to control in your system rather than guessing "Do I have under 10u barozine to stab myself with another space pen?".

**Comment by LankLTE** [comment]
> material reclaimer can already be used to extract chems from pens

Thought that was fixed but honestly don’t know.

I’m of the opinion that you shouldn’t be able to unload emergency / spacepens into a beaker.

**Comment by LankLTE** [comment]
> * injecting solutions into beakers from medipen is fine as long as reclaiming pens is fine

I don’t think reclaiming the pen is fine

**Comment by deltanedas** [comment]
ok make a pr fixing it

**Comment by VasilisThePikachu** [comment]
mfw

**Comment by UbaserB** [comment]
iirc barozine lets you survive in space. Fun fact, cryoxadone also can and it even heals you in the process. Because of this, i dont see any issue at all with barozine being obtainable.

**Comment by deltanedas** [comment]
bicaridine also exists just pop it and boom even works after youve been in space

**Comment by UbaserB** [comment]
exactly this

**Comment by LankLTE** [comment]
Neither Cryox or Bicar can fully outheal space damage- this is even when foams worked so I doubt it’s seriously viable now (as much as barozine, anyways). There’s also combat medipens and brute / burn autoinjectors that can be emptied

**Comment by deltanedas** [comment]
brute and burn injectors are literally:
- bicar and tranex
- derma and lepo

all of which are trivial to make

**Comment by LankLTE** [comment]
> the value of them are that they instantly inject

Yeah fair those are kinda awful, but I think keeping medipen contents inside medipens is a good precedent to set now

**Comment by VasilisThePikachu** [comment]
Even if we do decide to let medipens be able to inject into beakers. This commit should let you customize that easily. Or for anything in the feature.

**Comment by JoeHammad1844** [comment]
finally, no more 50u transfer jugs completely fucking with my hypo contents and making me OD people

**Comment by NancokPS2** [comment]
I feel like denying such a basic mechanic because it makes it so the one pen with an unobtainable (yet widely available) chem can have it extracted is very non-sensical and backwards.    

If barozine is so valuable, then it shouldn't be so widespread. If it isn't a problem to make it so widespread, then it isn't that valuable.    

The issue here is with the chem if anything, not the mechanic.  
  
Making pens be completely unmodifiable is just boring. Medical could prepare their own mixes or refill pens so they aren't completely disposable.  
Without mentioning that a pen is size 3 while a pill is size 1. So it isn't even like they are a direct upgrade (plus you still need an empty injector)

**Comment by LankLTE** [comment]
> I feel like denying such a basic mechanic because

Never said I would be denying it. I’m just stating what I think is an important change. 

> barozine shouldn't be a free space immunity either

This hasn’t been a problem ever, as barozine is super rare and only ever used once per round by most people.

File: Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@
   - type: ExaminableSolution
     solution: hypospray
   - type: Hypospray
+    onlyMobs: false
```
Comment (LankLTE) [inline Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml:21]: Add this to the Gorlex Hypospray (and probably the Hypopen too).

**Review by LankLTE — COMMENTED** [review]
Fairly minor, but the Syndicate Hyposprays should probably share this functionality.

File: Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml (hunk 1 above)
Comment (VasilisThePikachu) [reply Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml:21]: oh, I thought gorlax spray parented from hypospray. One moment

Comment (LankLTE) [reply Resources/Prototypes/Entities/Objects/Specific/Medical/hypospray.yml:21]: It does, but for some reason it reuses the hypospray component

**Review by LankLTE — APPROVED** [review]

**Pull request merged**